engine/runtime/src/main/java/org/enso/interpreter/runtime/state/ExecutionEnvironment.java

```diff
@@ -0,0 +1,77 @@
+package org.enso.interpreter.runtime.state;
+
+import com.oracle.truffle.api.CompilerDirectives;
+import org.enso.interpreter.node.expression.builtin.runtime.Context;
+import org.enso.interpreter.runtime.EnsoContext;
+import org.enso.interpreter.runtime.callable.atom.Atom;
+
+import java.util.HashMap;
+import java.util.Map;
+
+public class ExecutionEnvironment {
+
+  private final String name;
+  private final Map<String, Boolean> permissions;
+
+  public static final ExecutionEnvironment LIVE = initLive();
+  public static final ExecutionEnvironment DESIGN = new ExecutionEnvironment("design");
+
+  private static final ExecutionEnvironment initLive() {
+    Map<String, Boolean> contexts = new HashMap<>();
+    contexts.put("Input", true);
```

Comment:
Shouldn't these names be constants?

Reply:
Ideally we would get those from the builtin's constructor list except... that won't be initialized yet.
I can make it slightly less hacky, though.